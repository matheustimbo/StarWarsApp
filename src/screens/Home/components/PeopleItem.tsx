import React from 'react';
import {View, StyleSheet, Text, Dimensions} from 'react-native';
import {People} from '../../../providers/starWarsContext';

const {width} = Dimensions.get('window');

const PeopleItem: React.FC<{people?: People}> = ({people}) => {
  console.log('ta recebendo', people);
  return (
    <View style={styles.container}>
      <Text>Name: {people?.name}</Text>
      <Text>Height: {people?.height}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: width - 32,
    paddingHorizontal: 24,
    paddingVertical: 32,
    borderRadius: 8,
    justifyContent: 'space-between',
    backgroundColor: 'gray',
    marginVertical: 12,
    alignSelf: 'center',
  },
});

export default PeopleItem;
