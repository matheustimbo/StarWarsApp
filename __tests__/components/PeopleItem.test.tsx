import React from 'react';
import renderer from 'react-test-renderer';
import PeopleItem from '../../src/screens/Home/components/PeopleItem';

const examplePeople = {
  name: 'test',
  height: 180,
  mass: 50,
  hair_color: 'test',
  skin_color: 'test',
  eye_color: 'test',
  birth_year: 'test',
  gender: 'test',
  homeworld: 'test',
  films: ['test'],
  species: ['test'],
  vehicles: ['test'],
  starships: ['test'],
  url: 'test',
};

test('renders correctly when is bookmarked', () => {
  const tree = renderer
    .create(
      <PeopleItem
        people={examplePeople}
        goToDetails={() => {}}
        onPressBookmark={() => {}}
        isBookmarked={() => true}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly when isnÂ´t bookmarked', () => {
  const tree = renderer
    .create(
      <PeopleItem
        people={examplePeople}
        goToDetails={() => {}}
        onPressBookmark={() => {}}
        isBookmarked={() => false}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
