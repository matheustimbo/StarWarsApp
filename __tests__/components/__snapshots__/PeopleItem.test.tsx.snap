// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when is bookmarked 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#2F3B52",
      "borderRadius": 12,
      "elevation": 5,
      "justifyContent": "space-between",
      "marginVertical": 12,
      "paddingHorizontal": 24,
      "paddingVertical": 32,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.29,
      "shadowRadius": 6,
      "width": 718,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "position": "absolute",
        "right": 24,
        "top": 32,
        "zIndex": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#FFB84D",
            "fontSize": 24,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#98ABC9",
        "marginRight": 8,
        "marginTop": 8,
      }
    }
  >
    Name: 
    <Text
      style={
        Object {
          "color": "#fff",
          "textTransform": "uppercase",
        }
      }
    >
      test
    </Text>
  </Text>
  <Text
    style={
      Object {
        "color": "#98ABC9",
        "marginRight": 8,
        "marginTop": 8,
      }
    }
  >
    Gender: 
    <Text
      style={
        Object {
          "color": "#fff",
          "textTransform": "uppercase",
        }
      }
    >
      test
    </Text>
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FF951F",
          "fontWeight": "bold",
          "marginTop": 24,
          "textTransform": "uppercase",
        }
      }
    >
      See details
    </Text>
  </View>
</View>
`;

exports[`renders correctly when isn´t bookmarked 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#2F3B52",
      "borderRadius": 12,
      "elevation": 5,
      "justifyContent": "space-between",
      "marginVertical": 12,
      "paddingHorizontal": 24,
      "paddingVertical": 32,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.29,
      "shadowRadius": 6,
      "width": 718,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "position": "absolute",
        "right": 24,
        "top": 32,
        "zIndex": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#FFB84D",
            "fontSize": 24,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#98ABC9",
        "marginRight": 8,
        "marginTop": 8,
      }
    }
  >
    Name: 
    <Text
      style={
        Object {
          "color": "#fff",
          "textTransform": "uppercase",
        }
      }
    >
      test
    </Text>
  </Text>
  <Text
    style={
      Object {
        "color": "#98ABC9",
        "marginRight": 8,
        "marginTop": 8,
      }
    }
  >
    Gender: 
    <Text
      style={
        Object {
          "color": "#fff",
          "textTransform": "uppercase",
        }
      }
    >
      test
    </Text>
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FF951F",
          "fontWeight": "bold",
          "marginTop": 24,
          "textTransform": "uppercase",
        }
      }
    >
      See details
    </Text>
  </View>
</View>
`;
